docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+ 0
+ 0
+
+julia> s
+4-element Vector{Int64}:
+ 25
+  0
+  0
+ 56
+```
+
+Similar syntax is fully supported in DataFrames.jl. Here, Column `:Age` is replaced freshly allocated 
+vector because of broadcasting assignment:
+
+```jldoctest dataframe
+julia> df1[!, :Age] .= [85, 89, 78, 58, 96, 68] # col `:Age` is replaced freshly allocated vector
+6-element Vector{Int64}:
+ 85
+ 89
+ 78
+ 58
+ 96
+ 68
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    85  male             4
+   2 │    89  female           5
+   3 │    78  male             7
+   4 │    58  transgender      8
+   5 │    96  female           2
+   6 │    68  female           1
+```
+
+Note that if columns `:Customers` and `:City` are not present in `df1` then using `!` and `:` are equivalent. The 
+major difference between in-place and replace operations is that replacing columns is needed if new values have a 
+different type than the old ones:
+
+```jldoctest dataframe
+julia> df1[!, :Customers] .= ["Rohit", "Akshat", "Rahul", "Aayush", "Prateek", "Anam"] # allocates a new column `:Customers` and adds it
+6-element Vector{String}:
+ "Rohit"
+ "Akshat"
+ "Rahul"
+ "Aayush"
+ "Prateek"
+ "Anam"
+
+julia> df1[:, :City] .= ["Kanpur", "Lucknow", "Bhuvneshwar", "Jaipur", "Ranchi", "Dehradoon"] # allocates the column because `:City` is not present in `data`
+6-element Vector{String}:
+ "Kanpur"
+ "Lucknow"
+ "Bhuvneshwar"
+ "Jaipur"
+ "Ranchi"
+ "Dehradoon"
+
+julia> df1
+6×5 DataFrame
+ Row │ Age    Sex          Job    Customers  City
+     │ Int64  String       Int64  String     String
+─────┼───────────────────────────────────────────────────
+   1 │    85  male             4  Rohit      Kanpur
+   2 │    89  female           5  Akshat     Lucknow
+   3 │    78  male             7  Rahul      Bhuvneshwar
+   4 │    58  transgender      8  Aayush     Jaipur
+   5 │    96  female           2  Prateek    Ranchi
+   6 │    68  female           1  Anam       Dehradoon
+```
+
+Assignment of a scalar to a data frame can be done in ranges using broadcasting:
+
+```jldoctest dataframe
+julia> df1[:, 3] .= 4 # an in-place replacement of values stored in column number 3 by 4
+6-element view(::Vector{Int64}, :) with eltype Int64:
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+
+julia> df1
+6×5 DataFrame
+ Row │ Age    Sex          Job    Customers  City
+     │ Int64  String       Int64  String     String
+─────┼───────────────────────────────────────────────────
+   1 │    85  male             4  Rohit      Kanpur
+   2 │    89  female           4  Akshat     Lucknow
+   3 │    78  male             4  Rahul      Bhuvneshwar
+   4 │    58  transgender      4  Aayush     Jaipur
+   5 │    96  female           4  Prateek    Ranchi
+   6 │    68  female           4  Anam       Dehradoon
+```
+
+Here, `!` get us columns without copying and when setting columns it replaces them while 
```

Comment:
I do not see any difference between this example and the one above

Reply:
Are you pointing to this example:
```
julia> df1[:, :Age] .= "Economics"
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Int64

julia> df1[!, :Age] .= "Economics"
6-element Vector{String}:
 "Economics"
 "Economics"
 "Economics"
 "Economics"
 "Economics"
 "Economics"
```

Or, 

```
julia> df1[:, 3] .= 4 # an in-place replacement of values stored in column number 3 by 4
6-element view(::Vector{Int64}, :) with eltype Int64:
 4
 4
 4
 4
 4
 4

julia> df1
6×5 DataFrame
 Row │ Age    Sex          Job    Customers  City
     │ Int64  String       Int64  String     String
─────┼───────────────────────────────────────────────────
   1 │    85  male             4  Rohit      Kanpur
   2 │    89  female           4  Akshat     Lucknow
   3 │    78  male             4  Rahul      Bhuvneshwar
   4 │    58  transgender      4  Aayush     Jaipur
   5 │    96  female           4  Prateek    Ranchi
   6 │    68  female           4  Anam       Dehradoon
```

Reply:
Analyzing it again - I think it is OK.